utils/obsproc/NetCDFToIodaConverter.h

```diff
@@ -14,46 +14,106 @@
+
+      // ioda output file name
+      this->outputFilename_ = fullConfig.getString("output file");
+      oops::Log::info() << "--- Output files: " << this->outputFilename_ << std::endl;
+    }
+
+    // Method to write out a IODA file (writter called in destructor)
     void WriteToIoda() {
       // Create empty group backed by HDF file
       ioda::Group group =
         ioda::Engines::HH::createFile(
                                       this->outputFilename_,
                                       ioda::Engines::BackendCreateModes::Truncate_If_Exists);
-      this->ReadFromNetCDF(group);
+
+      // Extract ioda variables from the provider's files
+      std::string fileName = this->inputFilenames_[0];  // TODO(Guillaume): make it work for a list
+      gdasapp::IodaVars iodaVars;
+      this->ProviderToIodaVars(fileName, iodaVars);
+      oops::Log::debug() << "--- iodaVars.location: " << iodaVars.location << std::endl;
+      oops::Log::debug() << "--- iodaVars.obsVal: " << iodaVars.obsVal << std::endl;
+
+      // Update the group with the location dimension
+      ioda::NewDimensionScales_t
+        newDims {ioda::NewDimensionScale<int>("Location", iodaVars.location)};
+      ioda::ObsGroup ogrp = ioda::ObsGroup::generate(group, newDims);
+
+      // Set up the float creation parameters
+      ioda::VariableCreationParameters float_params;
+      float_params.chunk = true;               // allow chunking
+      float_params.compressWithGZIP();         // compress using gzip
+      float missing_value = util::missingValue(missing_value);
+      float_params.setFillValue<float>(missing_value);
+
+      // Create the IODA variables
+      ioda::Variable adtIodaDatetime =
+        ogrp.vars.createWithScales<float>("Metadata/dateTime",
+                                          {ogrp.vars["Location"]}, float_params);
+      // TODO(Mindo): Get the date info from the netcdf file
+      adtIodaDatetime.atts.add<std::string>("units", {"seconds since 9999-04-15T12:00:00Z"}, {1});
+
+      ioda::Variable adtIodaObsVal =
+        ogrp.vars.createWithScales<float>("ObsValue/"+this->variable_,
+                                          {ogrp.vars["Location"]}, float_params);
+      ioda::Variable adtIodaObsErr =
+        ogrp.vars.createWithScales<float>("ObsError/"+this->variable_,
+                                          {ogrp.vars["Location"]}, float_params);
+
+      // Write adt obs value to group
+      adtIodaObsVal.writeWithEigenRegular(*iodaVars.obsVal);
+
+      // Write adt obs error to group
+      adtIodaObsErr.writeWithEigenRegular(*iodaVars.obsError);
     }
 
    private:
-    // Virtual method to read a netcdf file and store the relevant info in a group
-    virtual void ReadFromNetCDF(ioda::Group &) = 0;
-
+    // Virtual method that reads the provider's netcdf file and store the relevant
+    // info in a IodaVars struct
+    virtual void ProviderToIodaVars(std::string fileName, gdasapp::IodaVars & iodaVars) = 0;
```

Comment:
should method start with lower case letter?

Reply:
:+1: